File: ios/Plugin/AppLanguagePlugin.m
#import <Foundation/Foundation.h>
#import <Capacitor/Capacitor.h>

// Define the plugin using the CAP_PLUGIN Macro, and
// each method the plugin supports using the CAP_PLUGIN_METHOD macro.
CAP_PLUGIN(AppLanguagePlugin, "AppLanguage",
           CAP_PLUGIN_METHOD(getApplicationLocales, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(setApplicationLocales, CAPPluginReturnNone);
           CAP_PLUGIN_METHOD(resetApplicationLocales, CAPPluginReturnNone);

           CAP_PLUGIN_METHOD(getSystemLocales, CAPPluginReturnPromise);

           CAP_PLUGIN_METHOD(getOverrideLocaleConfig, CAPPluginReturnNone);
           CAP_PLUGIN_METHOD(setOverrideLocaleConfig, CAPPluginReturnNone);

           CAP_PLUGIN_METHOD(openSettings, CAPPluginReturnNone);
)

File: ios/Plugin/AppLanguagePlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(AppLanguagePlugin)
public class AppLanguagePlugin: CAPPlugin {
    private let implementation = AppLanguage()

    @objc func getApplicationLocales(_ call: CAPPluginCall) {
        call.resolve([
            "locales": implementation.getApplicationLocales()
        ])
    }

    @objc func setApplicationLocales(_ call: CAPPluginCall) {
        call.unimplemented("Not implemented on iOS.")
    }

    @objc func resetApplicationLocales(_ call: CAPPluginCall) {
        call.unimplemented("Not implemented on iOS.")
    }

    @objc func getSystemLocales(_ call: CAPPluginCall) {
        call.resolve([
            "locales": implementation.getSystemLocales()
        ])
    }

    @objc func getOverrideLocaleConfig(_ call: CAPPluginCall) {
        call.unimplemented("Not implemented on iOS.")
    }

    @objc func setOverrideLocaleConfig(_ call: CAPPluginCall) {
        call.unimplemented("Not implemented on iOS.")
    }
}
